modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskServiceImpl.java

```diff
@@ -525,7 +527,7 @@ private TaskDataProvider findProviderByName(DataSource ds) {
             errors.addAll(taskValidator.validateAction(action, channel));
             TriggerEvent trigger = triggerEventService.getTrigger(task.getTrigger());
             Map<Long, TaskDataProvider> providers = getProviders(task);
-            ActionEvent actionEvent = channel.getAction(action);
+            ActionEvent actionEvent = ChannelUtils.getAction(channel, action);
```

Comment:
Is there a reason for moving this kind of methods to the utils? Simple methods like contains or getAction fit well with the domain classes in my opinion. If we move all logic to static helpers, that will not be very object oriented.